### PR TITLE
Patch ElasticSearch deprecation warning

### PR DESCRIPTION
**Pull request by Bhacaz**
Hi, since some peoples seems to work/wish to make Chewy compatible with ElasticSearch 6-7, here is a small PR to fix the deprecation warning from ElasticSearch. It seems to be a good first step.

### CODE REVIEW

**Review by AlfonsoUceda — APPROVED** [review]

File: spec/chewy/index/actions_spec.rb [hunk 1]
```diff
@@ -55,7 +55,7 @@
     context do
       before { DummiesIndex.create }
       specify do
-        expect { DummiesIndex.create! }.to raise_error(Elasticsearch::Transport::Transport::Errors::BadRequest).with_message(/index_already_exists_exception.*dummies/)
+        expect { DummiesIndex.create! }.to raise_error(Elasticsearch::Transport::Transport::Errors::BadRequest).with_message(/already exists.*dummies/)
```
Comment (rabotyaga) [inline spec/chewy/index/actions_spec.rb:58]:
```suggestion
        expect { DummiesIndex.create! }.to raise_error(Elasticsearch::Transport::Transport::Errors::BadRequest).with_message(/resource_already_exists_exception.*dummies/)
```
https://github.com/elastic/elasticsearch/blob/feab123ba400b150f3dcd04dd27cf57474b70d5a/rest-api-spec/src/main/resources/rest-api-spec/test/indices.rollover/10_basic.yml#L131

File: spec/chewy/index/actions_spec.rb [hunk 2]
```diff
@@ -67,7 +67,7 @@
       specify { expect(DummiesIndex.aliases).to eq([]) }
       specify { expect(DummiesIndex.indexes).to eq(['dummies_2013']) }
       specify do
-        expect { DummiesIndex.create!('2013') }.to raise_error(Elasticsearch::Transport::Transport::Errors::BadRequest).with_message(/index_already_exists_exception.*dummies_2013/)
+        expect { DummiesIndex.create!('2013') }.to raise_error(Elasticsearch::Transport::Transport::Errors::BadRequest).with_message(/already exists.*dummies_2013/)
```
Comment (rabotyaga) [inline spec/chewy/index/actions_spec.rb:70]:
```suggestion
        expect { DummiesIndex.create!('2013') }.to raise_error(Elasticsearch::Transport::Transport::Errors::BadRequest).with_message(/resource_already_exists_exception.*dummies_2013/)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 # master
 
-  * Update gemspec dependencies for Rails. Update CI gemfiles and matrix to tests against current LTS Rails versions.
+  * Update gemspec dependencies for Rails. Update CI gemfiles and matrix to tests against current LTS Rails versions. (@bhacaz, #733)
+
+  * Correct deprecation warning for Elasticsearch 5.6 to 6: empty query for`_delete_by_query`, delete by alias (@bhacaz, )
```
Comment (rabotyaga) [inline CHANGELOG.md:5]:
```suggestion
  * Correct deprecation warning for Elasticsearch 5.6 to 6: empty query for`_delete_by_query`, delete by alias, `index_already_exists_exception` renaming (@bhacaz, #735)
```

**Review by rabotyaga — APPROVED** [review]
Hey @Bhacaz 👋 !
Thanks a lot!
Please take a look at these suggestions.

File: spec/chewy/index/actions_spec.rb (hunk 1 above)
Comment (rabotyaga) [reply spec/chewy/index/actions_spec.rb:58]: Discard this. Otherwise, it will stop working on ES5.

**Review by mrzasa — APPROVED** [review]

File: spec/chewy/index/actions_spec.rb (hunk 1 above)
Comment (Bhacaz) [reply spec/chewy/index/actions_spec.rb:58]: Just to keep track of the differences:

```
ES 5 [400] {"error":{"root_cause":[{"type":"index_already_exists_exception","reason":"index [dummies/xadaxSc9QhONQCws2qcZzw] already exists","index_uuid":"xadaxSc9QhONQCws2qcZzw","index":"dummies"}],"type":"index_already_exists_exception","reason":"index [dummies/xadaxSc9QhONQCws2qcZzw] already exists","index_uuid":"xadaxSc9QhONQCws2qcZzw","index":"dummies"},"status":400}
ES 6 [400] {"error":{"root_cause":[{"type":"resource_already_exists_exception","reason":"index [dummies/nmx0XrJlQi6so8rl_m4dnw] already exists","index_uuid":"nmx0XrJlQi6so8rl_m4dnw","index":"dummies"}],"type":"resource_already_exists_exception","reason":"index [dummies/nmx0XrJlQi6so8rl_m4dnw] already exists","index_uuid":"nmx0XrJlQi6so8rl_m4dnw","index":"dummies"},"status":400}
 ```

**Review by rabotyaga — APPROVED** [review]

**Pull request merged**